### DISCUSSION TITLE
How to customize top tabs' bar indicator with animation?

### DISCUSSION THREAD

**Discussion by trojanyao** [react-navigation/react-navigation#12037] [General]
I follow the document with [`tabbarIndicator`](https://reactnavigation.org/docs/material-top-tab-navigator/#tabbarindicator) option to customize the bar indicator element. But the customized indicator has no animation when changing index, it's move instantly.

How can I add animation like the original moving effect. The document about `tabbarIndicator` option is very simple.

<img width="597" alt="image" src="https://github.com/react-navigation/react-navigation/assets/22811809/0d51793d-4af6-46c2-a2e6-2cee50c59bc4">

I tried the following way to animate the `translateX` prop (in Expo's `_layout` file), but it not works. There's many strange errors in console.

<img width="1019" alt="image" src="https://github.com/react-navigation/react-navigation/assets/22811809/094cd15f-9005-4d4f-a9e4-5c00d30b4e65">
<img width="1926" alt="image" src="https://github.com/react-navigation/react-navigation/assets/22811809/1118a33e-1582-495a-af68-c3bda39d0136">

**Comment by satya164** [comment]
https://github.com/react-navigation/react-navigation/blob/5e297a05d7457336ef8f8d1bddaa486b5d8ab4e1/example/src/Screens/TabView/CustomIndicator.tsx#L51
